lib/services/lookup.js

```diff
@@ -482,6 +482,31 @@ function lookupServiceFactory(
     };
 
     /**
+     * Returns a list of object containing IP and Mac Addresses for the given Node ID.
+     * @param  {String} id Node ID
+     * @return {Promise.<Array>} A promise fulfilled with an array of
+     * objects containing IP Addresses and Mac Addresses associated with the Node.
+     */
+    LookupService.prototype.findIpMacAddresses = function (id) {
+        var self = this;
+        assert.isMongoId(id);
+
+        return self.validateArpCache()
+        .then(function() {
+            return waterline.lookups.findByTerm(id).then(function (records) {
+                return _.reduce(records, function (ipMacAddresses, record) {
+		    var ipMacAddress = {};
+	            ipMacAddress.ipAddress = record.ipAddress;
+		    ipMacAddress.macAddress = record.macAddress;
+                    ipMacAddresses.push(ipMacAddress);
```

Comment:
Just a little coding suggestion, you could avoid to explicitly declare a varaible `ipMacAddress`, just do something like below:
```
ipMacAddresses.push({
    ipAddress: record.ipAddress,
    macAddress: record.macAddress
});
```

or event a more simple one:
```
ipMacAddresses.push(_.pick(record, ['ipAddress', 'macAddress']));
```